synapse/storage/databases/main/room.py

```diff
@@ -875,6 +899,46 @@ def _quarantine_media_txn(
 
         return total_media_quarantined
 
+    def _remove_from_quarantine_media_txn(
+        self,
+        txn,
+        local_mxcs: List[str],
+        remote_mxcs: List[Tuple[str, str]],
+    ) -> int:
+        """Remove from quarantine local and remote media items
+
+        Args:
+            txn (cursor)
+            local_mxcs: A list of local mxc URLs
+            remote_mxcs: A list of (remote server, media id) tuples representing
+                remote mxc URLs
+        Returns:
+            The total number of media items removed from quarantine
+        """
+        # Update all the tables to set the quarantined_by flag to NULL
+        txn.executemany(
+            """
+            UPDATE local_media_repository
+            SET quarantined_by = ?
+            WHERE media_id = ?
+        """,
+            ((None, media_id) for media_id in local_mxcs),
+        )
+        # Note that a rowcount of -1 can be used to indicate no rows were affected.
+        total_media_remove_quarantine = txn.rowcount if txn.rowcount > 0 else 0
+
+        txn.executemany(
+            """
+                UPDATE remote_media_cache
+                SET quarantined_by = ?
+                WHERE media_origin = ? AND media_id = ?
+            """,
+            ((None, origin, media_id) for origin, media_id in remote_mxcs),
+        )
+        total_media_remove_quarantine += txn.rowcount if txn.rowcount > 0 else 0
+
+        return total_media_remove_quarantine
```

Comment:
I think the part for discussion in this PR is this. This is nearly a clone of `_quarantine_media_txn` only without parameter `quarantined_by` an a SQL query without `AND safe_from_quarantine = ?`
https://github.com/matrix-org/synapse/blob/c89da79155013b802899c2e5d3a53ba993972c22/synapse/storage/databases/main/room.py#L860-L900

I can change it to one function if you like. Then I will make parameter `quarantined_by` to `Optional[str]`

Reply:
well, I did wonder about that, but didn't want to insist that you change it.

Yes, I think that combining the functions would be an improvement, if you have time to do so. Thanks.

Reply:
I have made an update.